api/v1alpha1/hardwareclassification_types.go

```diff
@@ -27,8 +24,61 @@ type HardwareClassificationSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of HardwareClassification. Edit HardwareClassification_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// ExpectedHardwareConfiguration defines expected hardware configurations for CPU, RAM, Disk, NIC.
+	ExpectedHardwareConfiguration ExpectedHardwareConfiguration `json:"expectedValidationConfiguration"`
+}
+
+// ExpectedHardwareConfiguration details to match with the host
+type ExpectedHardwareConfiguration struct {
+	Namespace string `json:"namespace"`
```

Comment:
Every CRD includes metadata that has the namespace. It seems unusual to ask for that twice, or to allow the namespace for a HardwareClassification to be different from the one to which it applies.

Reply:
Yes this parameter is removed and we are using namespace from metadata.
Addressed in new commit: c239dc6